### DISCUSSION TITLE
ggplot2 extensions session at JSM 2025?

### DISCUSSION THREAD

**Discussion by jtr13** [ggplot2-extenders/ggplot-extension-club#64] [General]
Hi All, 

I'm posting to see if there's interest in speaking at a Statistical Graphics sponsored [topic contributed session](https://ww2.amstat.org/meetings/jsm/2025/topiccontributed.cfm) that Gina and/or I would organize at JSM 2025 in Nashville (Aug 2-7) on ggplot2 extensions. The timing is ideal as Hadley will be receiving the [ASA Statistical Computing and Graphics Award](https://posit.co/blog/hadley-wickham-honored-with-the-2025-asa-award-for-statistical-computing-and-graphics/) at the conference with a session in his honor, not to mention that Nashville is high on the conference-location-desirability list. 

To pull it off, we would need a minimum of three speakers, ideally four or five. 

**The timeline**:  The deadline for submitting a proposal is December 9, at which time we would only need a list of speakers and tentative talk titles. If the proposal is accepted -- I believe there is a very high chance that it would be -- speakers would need to submit abstracts by February 3, 2025, as well as pay a $100 submission fee which is in essence a registration deposit.

**The economic reality**: JSM is not cheap. The early registration cost -- before June 3 -- for members of sponsoring organizations (ASA, CAS, CWS, ENAR, ICSA, IISA, IMS, ISBA, ISI, KISS, RSS, SSC, WNAR, SSA) is $545 and for non-members $820 ([full registration fee chart here](https://ww2.amstat.org/meetings/jsm/2025/registration.cfm)). To be clear that does not include travel, lodging, meals, etc.

If you are potentially interested, please comment below with the package or topic you'd be interested in speaking about and roughly how likely you are to be able to attend.

I have organized graphics sessions at JSM in the past (@corybrunson spoke at one on [ordr](https://github.com/corybrunson/ordr)) and they have been well-attended. Statisticians interested in graphics are an ideal audience for ggplot2 extension packages. I really believe such a session would be a valuable contribution to the conference program.

This group has priority for speaker slots until Sunday night 11/24!

best,
Joyce

**Comment by corybrunson** [comment]
I am interested and would almost certainly attend! Since i've already presented {ordr}, i could present either of the following instead:
* [{ggalluvial}](https://github.com/corybrunson/ggalluvial/) – I've actually never presented this package to a forum before, and i have some unpopular opinions about alluvial and related graphics that it might be helpful to have openly challenged.
* [{ggtda}](https://github.com/tdaverse/ggtda) – This is joint work primarily with @rrrlw and @jamesotto852 that does some clever things with {ggplot2} internals and is part of a larger project to make modular topological data analytic tools available to R and (but not only) Tidyverse users.

**Reply by jtr13** [reply]
Yay!

**Comment by corybrunson** [comment]
Also—may we tag others outside the extensions group who might be interested?

**Reply by teunbrand** [reply]
This is up to Joyce, I have no personal objections

**Reply by EvaMaeRey** [reply]
given the looming deadline, maybe tagging folks that come to mind is the way to go - especially if JSM might be in their plans like @corybrunson! (and then they also join us sometime!)  @jtr13?

**Reply by jtr13** [reply]
I'd like to give preference to this group so let's give it until Sunday and then start tagging others. @EvaMaeRey Will everyone in the group see this message?

**Reply by EvaMaeRey** [reply]
Sounds good! I will include a link to this thread in the ggswim invite to give this group a heads up on the opportunity! I'll send that out tonight or tomorrow morning 🌄

**Reply by jtr13** [reply]
Thanks @EvaMaeRey! It would be great if you could mention that this group has priority for speaker slots until **Sunday night 11/24**.

**Reply by EvaMaeRey** [reply]
I sent out the email.  I edited your intial post with 'this group has priority for speaker slots until Sunday night 11/24.' which felt a little weird to do, but it seemed like it fit there better than in the email!

**Comment by yjunechoe** [comment]
That sounds like a super fun session!

My very selfish proposal for a talk (merely tangentially related to ggplot _extensions_, so please only consider if you need more speakers) is to give my ["Sublayer Modularity in the Grammar of Graphics"](https://www.youtube.com/watch?v=613Q0j6Kjm0) talk in person 😅 

The context is that I was scheduled to give this talk at JSM 2023 in the student paper award session organized by the sections on Statistical Computing and Statistical Graphics. I unfortunately got covid right before the trip, so I sent in a recording of the talk, but it didn't end up getting played due to issues w/ tech setup (IIRC it just got shared around unofficially). So it's maybe technically still new for the audience(?)

I've also always wanted to do a variant of the talk where I make it wholly about ggplot2. Ex: in earlier drafts I entertained some quirky titles like "ggplot2 and lazy evaluation: a love story" and stuff like that, which I eventually discarded in favor of focusing on the bigger, theoretical picture (of Grammar of Graphics) 😆 

Anyways, just an idea for your back pocket! I'm also actually not certain whether I can attend (I should probably know around Jan~Feb)

**Reply by jtr13** [reply]
I think the topic sounds perfect! And congratulations on the award!

**Reply by yjunechoe** [reply]
thank you! 😃

**Comment by mjskay** [comment]
This sounds like it could be fun! If it's of interest I could give a talk on [ggdist](https://mjskay.github.io/ggdist/) --- I think for the JSM audience I could also touch on how [ggdist](https://mjskay.github.io/ggdist/) fits into a broader ecosystem of packages for working with distributions in R (like [distributional](https://pkg.mitchelloharawild.com/distributional/) and [posterior::rvar](https://mc-stan.org/posterior/articles/rvar.html)) and how it tries to formally integrate distributions into the grammar of graphics.

I have funding I can use for travel and I don't have anything else at that time, so it is likely I would be able to attend.

**Reply by jtr13** [reply]
Fantastic! Definitely of interest!

**Comment by tdhock** [comment]
I may go to JSM and if so could present about directlabels or animint2

**Reply by jtr13** [reply]
Great news!

**Reply by jtr13** [reply]
The deadline is coming up soon so if you're still interested could you post or email the info needed for the JSM submission [here](https://github.com/teunbrand/ggplot-extension-club/discussions/64#discussioncomment-11385899)? Thanks!

**Comment by EvaMaeRey** [comment]
Joyce and I will work on a talk like: 'Who are the ggplot2 extenders? (and how to become one)'.   Open to ideas about how to look at this question!

**Reply by corybrunson** [reply]
It might be difficult to acquire relevant data, but, if i could have one version of this question answered, then it might be how extenders integrate this work into their professional duties or portfolios—so, indirectly, what jobs or roles tend to lead/allow people to become extenders. It's hard enough to justify software development as an academic researcher, let alone an extension to existing software, and yet this work led to my most-cited journal article, so i've long been struck by the issue.

**Reply by EvaMaeRey** [reply]
Thanks for this idea Cory!  This is a great question that I think we see touched upon here and there - so probably of interest to people. If you have specific ideas about how to answer this, please let us know!    

I feel like @jamesotto852, @larmarange have commented on some of these issues as well in their discussions, and might have more to add.

The whole 'who are extenders?' is a huge topic!  😬😅

**Reply by mjskay** [reply]
I love this question! Two (I think common) categories I've been in are: "PhD student working on stuff on the side" and "faculty working on stuff on the side". I have also occasionally been able to allocate a portion of summer salary from a grant to ggdist work, but the work I do in the summer is pretty unrestricted.

Like @corybrunson some of my most-cited stuff is R packages (including tidybayes, the predecessor to ggdist), which suggests it maybe shouldn't be treated as a side gig!

**Reply by teunbrand** [reply]
I definitely fell in the 'PhD student working on side stuff in my spare time' categorisation of this. I never contemplated publishing my extension packages anywhere else than CRAN because I don't know of any good venues to do so. I can see ggdist being tied up in stats work more closely than everybody's misc extension packages.

**Reply by EvaMaeRey** [reply]
I think it's probably worth starting a new thread to look at some of the different dimensions of this question, including the one Cory brought up. @jtr13 and I also want to look at extensions type, which I guess is more what extensions do than who are the extenders, but maybe reflects extenders' philosophy and toolsets.  There's some discussion in #62 from package imports and exports perspective.

**Reply by EvaMaeRey** [reply]
My extension work is not yet CRAN or academic-journal published. Publishing an article does seem to be the-credit claiming path for academics. And usually there's interesting theoretical ideas that motivate extension (even within misc packages!) so highlighting/uncovering paths to publishing seems pretty worthwhile discussion...  Maybe a starting point would be looking at what packages have paper citation information could help gather info ... citation(gg*)

**Reply by EvaMaeRey** [reply]
Then, special meetup on publishing experiences!

**Comment by jtr13** [comment]
So... we potentially have a full panel 🎉 🎉 🎉 !

**yes**
1. Gina @EvaMaeRey / Joyce @jtr13 
2. Cory @corybrunson 
3. Matthew @mjskay 

**maybe**
1. June  @yjunechoe 
2. Toby  @tdhock 

(Note: Any of the talks can have multiple authors who don't have to be in attendance.)

@corybrunson : Up to you how you want to handle spreading the word... we could try to find backup speakers now (who could submit as individuals to a [contributed session](https://ww2.amstat.org/meetings/jsm/2025/contributedsessions.cfm) up to Feb 3 if we can't accommodate them) or we could wait and look for replacements on an as needed basis.

**Reply by corybrunson** [reply]
@jtr13 that makes sense from an organizing perspective. I worry that i'll forget, and i always appreciate advance notice, so i'll tag a couple of folks in this thread as a way to alert them to the idea, and they can comment if they are interested. Thank you so much!

**Reply by jtr13** [reply]
Sounds good, thank you!

**Reply by mjskay** [reply]
Cool! Do you need any additional info from us now?

**Reply by jtr13** [reply]
Great question! I always think I have everything I need and then scramble at the deadline to find out how speakers like their names to appear and which email addresses to use. So, please post (or email me --  jtr13@columbia.edu -- and @EvaMaeRey whose email address you should have) the following let's say by Dec 5:

@yjunechoe and @tdhock can you provide this info? Thanks!

**required**
Tentative talk title
First and last name as you want it to appear in the program
Affiliation
Email

**optional**
job title
mailing address

Thank you!!

**Reply by jtr13** [reply]
Also, just realized that listing a **chair** for the session is required, and it cannot be an organizer or speaker. The job is to introduce the speakers and warn them when time is up. Is there anyone who will be attending JSM who would like this position or who knows someone who would? Thanks!

**Reply by mjskay** [reply]
Thanks! For me you can use:

ggdist: Visualizations of distributions and uncertainty in the grammar of graphics
Matthew Kay
Associate Professor, Computer Science and Communication Studies
Northwestern University
matthew.kay@u.northwestern.edu

**Reply by jtr13** [reply]
Thanks!

**Reply by corybrunson** [reply]
A grammar and taxonomy of ribbon graphics
Jason Cory Brunson
Assistant Professor
Laboratory for Systems Medicine, University of Florida
P.O. Box 100225 JHMHC, Gainesville FL 32610-0225
jason.brunson@medicine.ufl.edu

**Reply by yjunechoe** [reply]
(Sorry -- I completely missed this thread!)

Towards a sub-layer grammar of graphics
June Choe
Ph.D. Candidate
Department of Linguistics, University of Pennsylvania
yjchoe@sas.upenn.edu

**Reply by jtr13** [reply]
No worries, thanks!

**Reply by cynthiahqy** [reply]
**Title:** Examining ggplot2 extensions for grammar of graphics design patterns
**Name**: Cynthia A. Huang, Mitchell O'Hara-Wild
**Affiliation:** Monash University
**email:** cynthia.huang@monash.edu

**Reply by cynthiahqy** [reply]
@jtr13 I'd be happy to chair if needed (and a spot doesn't open up for my talk)

**Reply by jtr13** [reply]
Thanks!

**Reply by EvaMaeRey** [reply]
I think there might be some synergies between the 'who are the extenders' and the 'design patterns' talks.   I wonder if it would be worth comparing notes.  @jtr13 has proposed looking at taxonomy, especially w/ regard to the prescribed extension points.  In #62, which initially points to an effort which I think looks more at subject area typology of extension package, but later in the thread we start to look at some exported functions too - but just packages from https://exts.ggplot2.tidyverse.org/.  Looking at packages based on imports is suggested by @yjunechoe there too - had not occurred to me (I'm in the barbie-dplyr camp! 😬)

**Comment by corybrunson** [comment]
@jamesotto852 would this interest you? For example, would you want to present {ggtda}? If not, and if i do, then i hope i could work with you to prepare the presentation.

**Comment by cynthiahqy** [comment]
@jtr13 this is such a great idea! I'm sorry to have missed this thread while travelling. It might be a little too late but I have a talk to offer (perhaps as a back-up). 

The talk would be on work with Mitchell O'Hara Wild analysing the source code of ggplot2 extensions packages. The current working title is "Examining ggplot2 extensions for grammar of graphics design patterns". 

I should have funding to attend JSM and could also submit this as a contributed talk instead if this panel is full.

**Reply by jtr13** [reply]
Wow, sounds great! We do have a tentative full panel now but a space could certainly open up. Could you post or email the info listed [here](https://github.com/teunbrand/ggplot-extension-club/discussions/64#discussioncomment-11385899) just in case?

**Reply by mjskay** [reply]
@cynthiahqy seeing your talk title intrigued me and led me to find [your related blog post](https://www.cynthiahqy.com/posts/ggplot-helper-design) as well as @mitchelloharawild's [related blog post](https://mitchelloharawild.com/blog/ggplot2-package-design/), both of which I enjoyed.

Awhile back I wrote a snarky take on this titled [*I don’t want your monolithic ggplot function*](https://blog.mjskay.com/2017/11/05/i-don-t-want-your-monolithic-ggplot-function/). Your posts struck me as being in perhaps a similar spirit but more productive tone :).

**Reply by corybrunson** [reply]
My submission would be more of a case study, so i'd be glad to transfer my panel seat to someone offering a deeper look at the broader extension ecosystem. If a contributed session were put together, how would it be connected to the panel?

**Reply by jtr13** [reply]
@corybrunson That's very generous but not necessary. I just figured out that if we switch from a topic contributed _paper_ session to a topic contributed _panel_ session we can have up to 6 panelists. The difference between the two isn't that great since panelists can give presentations at a panel session. In fact the session at which you spoke in 2022 was a panel session. It is probably a better fit anyway since the focus is more on discussion rather than formalistic individual research. For the panel session speakers do not need to submit individual abstracts for example. If we end up with fewer than 6 speakers we can discuss and perhaps switch to a paper session depending on the preferences of the group.

**Reply by corybrunson** [reply]
@jtr13 nice! Thank you again for all the effort you're putting into this.

**Comment by jtr13** [comment]
A quick note to let you know that the proposal is in! Thanks to everyone for your interest and enthusiasm. The decision date is "early January" -- of course I'll let you know as soon as I hear.

**Reply by EvaMaeRey** [reply]
Thanks for all the coordination work you've put into this, Joyce!  🙏🤞

**Comment by EvaMaeRey** [comment]
Just thought @jtr13  should be thanked here too - fun to take a quick look back and see where the planning began - last November!!  Thank you the idea and all of the coordination work you did Joyce!

**Reply by teunbrand** [reply]
For those of us that weren't there: what did we miss?

**Reply by EvaMaeRey** [reply]
IRL conversations would be the big thing! 🤷‍♀️  I guess we can share photos at the next meetup - maybe release party?  Looks like the release is imminent-ish?   We did talk about how different folks might contribute to discussing implications (and excitement for) of new release features at such an event.  :-)

**Reply by jtr13** [reply]
This definitely doesn't capture it but for the archival record, here's a link to the session info: https://ww3.aievolution.com/JSMAnnual2025/Events/viewEv?ev=2740

**Reply by EvaMaeRey** [reply]
We had a good turnout for the panel Tuesday. All scheduled speakers @mjskay, @corybrunson, @yjunechoe, @jtr13, @mitchelloharawild, me made it, and pretty great reception w/ folks like @heike, @srvanderplas, @dkahle and @huizezhang-sherry, who have each made it to our virtual meetings at some point and other folks like @zabore @jonkeane and @symanzik in attendance. Hopefully we'll meet more folks that were in attendance in the future.  I think most of us were at @hadley's graphics award and panel too - very nice event and presentations!

**Reply by EvaMaeRey** [reply]
Cross linking to @jyuu's LinkedIn post on the graphics award panel Wednesday! https://www.linkedin.com/posts/joycecahoon_had-a-great-time-celebrating-hadley-wickham-activity-7358976153346326529-gwAF?utm_source=share&utm_medium=member_desktop&rcm=ACoAAAIFK0wBe0Q89iD-ZEWren9Fjl9ugLy3zB0

**Reply by jyuu** [reply]
This is amazing!! I'm so sorry to have missed the ggplot panel, would absolutely loved to attend. So grateful to you all for making it out to the celebration, would love to connect with everyone in person in future as well

**Reply by EvaMaeRey** [reply]
Thank you for saying hi @jyuu!  🤩  Your MC skills made the atmosphere so celebratory! 🎉  You are always welcome at the extenders meetup... 📊🙃 [website](https://ggplot2-extenders.github.io/ggplot-extension-club/), [get in touch](https://docs.google.com/forms/d/e/1FAIpQLSe3M1KwUPrmTfEGuuQp0fZ0J7dZkk_82gb310JCvdouMTa_7Q/viewform)

**Comment by jtr13** [comment]
Or maybe a folder to share slides?

**Reply by EvaMaeRey** [reply]
Yeah, maybe it is time to start a new thread - like my slides are online (as you well know 😅), I'd be happy to link to them.

**Reply by corybrunson** [reply]
I've gotten some emails from ASA about submitting a manuscript to the proceedings. Is this something that makes sense for the lot of us to do, perhaps together? I'm not itching to write but i'm happy to join in.

**Reply by EvaMaeRey** [reply]
I guess my inclination would be for folks to decide whether to submit or not, but I may not be familiar with all the benefits afforded by posting to JSM's portal but maybe you @corybrunson, @srvanderplas or @jtr13 could speak to this?    I'd consider linking to this kind of thing from the extenders website (paper, slides) if a good number of folks were up for it.  My slides are here btw: https://evamaerey.github.io/mytidytuesday/2023-09-26-cowy-outline/extenders-jsm

**Reply by corybrunson** [reply]
Oops—i just realized i'd never made mine public. Here it is: https://github.com/corybrunson/2025-08-05-ggalluvial

**Reply by corybrunson** [reply]
From a crude professional standpoint, conference proceedings are low-cost, low-reward citations; they aren't (re)viewed as highly as journal articles, but they do count for something. If it could serve an additional purpose that isn't being served elsewhere, at least not for the same community, as @srvanderplas suggests, then i would be more inclined to do it!

**Reply by EvaMaeRey** [reply]
Thanks for the feedback, Cory!  I think it's reasonable then that authors make their own choice in this regard, as if submitting to the proceedings is worth it.

**Reply by corybrunson** [reply]
Thanks @EvaMaeRey, though i'm actually not myself sure about the preferred (by ASA) approach to proceedings submissions for panels, as opposed to contributeds or inviteds. Maybe @jtr13 has insight on this?

**Comment by EvaMaeRey** [comment]
Hadley's talk was mostly about llms, but brought in ggplot2 in a round about way, given as ggplot2 is used in the shiny apps that LLMs generate from an app sketch... 

<img width="335" height="223" alt="image" src="https://github.com/user-attachments/assets/57639353-aabf-4574-883e-99269cd4f986" />

<img width="777" height="495" alt="image" src="https://github.com/user-attachments/assets/38186326-609c-411b-9318-d826eeaa194a" />

**Reply by EvaMaeRey** [reply]
@mjskay, the filtering (species) step brought your study of how people use ggplot2 to mind - I think filtering gets mentioned in there.  I will link to it when I get the chance to search for it again!

**Comment by srvanderplas** [comment]
It would be a great place for a general how to guide with tips!!
